client/utils/password-toggle.js

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import propTypes from "prop-types";
+
+class PasswordToggleIcon extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isVisible: false,
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  componentDidMount() {
+    const {isVisible} = this.props;
+    this.setState({
+      isVisible,
+    });
+  }
+
+  handleClick(inputRef) {
+    const {isVisible} = this.state;
+    this.setState(
+      {
+        isVisible: !isVisible,
+      },
+      () => {
+        inputRef.current.setAttribute(
+          "type",
+          inputRef.current.getAttribute("type") === "password"
+            ? "text"
+            : "password",
+        );
+      },
+    );
+  }
+
+  render() {
+    const {inputRef, parentClassName} = this.props;
+    const showPasswordIcon = () => <i className="eye" />;
+    const hidePasswordIcon = () => <i className="eye-slash" />;
```

Comment:
can you please add a `title` attribute so that if the user leaves the mouse on the icon a hint will appear?

Eg:

- reveal password
- hide password

The only problem is how to make this strings translatable, with the current system we'd have to define it in the YAML, maybe defining it just once and passing that value to the component is enough for now, but for sure soon we'll have to start working on the [translation issue](https://github.com/openwisp/openwisp-wifi-login-pages/issues/235).